### DISCUSSION TITLE
How to switch languages without language setting

### DISCUSSION THREAD

**Discussion by louies0623** [helloSystem/hello#354] [General]
I see that the interface language of your preview image is German. I want to know how you switched it, because the language in our system has always been English, not the language you need to display.

**Comment by probonopd** [comment]
I attach an original [German Raspberry Pi Keyboard and Hub](https://www.raspberrypi.com/products/raspberry-pi-keyboard-and-hub/). It autodetects the keyboard language and sets the system language automatically according to it.
Alternatively, on Macs it reads an EFI variable that macOS sets and sets the keyboard language and system language automatically. Eventually we will be able to write that variable, too.

**Comment by louies0623** [comment]
Oh I see. 
But , Is there an instruction to switch the system language with the terminal finger?

**Reply by probonopd** [reply]
Currently not, you would have to edit the `/usr/local/sbin/localize` script.

**Reply by louies0623** [reply]
How to edit the script, can you give me the steps?

**Reply by probonopd** [reply]
I have not tested this: Delete all its code. Then edit `/usr/local/var/localize/include` to what you need. (Don't do this on an important work machine. Only on a spare system. No warranties!)

**Reply by louies0623** [reply]
By the way, open as root in tabmenu's too option doesn't work.

**Reply by louies0623** [reply]
I tried to change it to zh_TW and it turned back to en_US after rebooting.

**Reply by probonopd** [reply]
Then you still have code in `/usr/local/sbin/localize`?

**Reply by louies0623** [reply]
O god is working!
Even localize has to be changed, I started to think that a language setting is really necessary.

**Reply by louies0623** [reply]
It's just that some places need to be translated. ![VirtualBox_PC-2_04_04_2022_03_07_18](https://user-images.githubusercontent.com/44593430/161444054-d42d583f-9b63-4698-a355-b7f78d8ab1e8.png)

**Comment by louies0623** [comment]
Can you put it in the user manual because definitely someone will need it.

**Comment by 1cupoverflow** [comment]
I am trying to use helloSystem from the Live USB since I have no storage and I cannot seem to get the language right.
